gateway/api.go

```diff
@@ -2087,6 +2087,20 @@ func ctxGetOrigRequestURL(r *http.Request) *url.URL {
 	return nil
 }
 
+func ctxSetUrlRewriteTarget(r *http.Request, url *url.URL) {
```

Comment:
`ctxSetURLRewriteTarget` same for other new added `Url`s(not existing ones) throughout the PR.